ReactSkia/components/RSkComponentImage.cpp

```diff
@@ -253,9 +269,18 @@ void RSkComponentImage::requestNetworkImageData(ImageSource source) {
 
   // completioncallback lambda fuction
   auto completionCallback =  [&](void* curlresponseData,void *userdata)->bool {
+    auto component = getComponentData();
+    auto imageEventEmitter = std::static_pointer_cast<ImageEventEmitter const>(component.eventEmitter);
     CurlResponse *responseData =  (CurlResponse *)curlresponseData;
     CurlRequest * curlRequest = (CurlRequest *) userdata;
-    processImageData(responseData->responseurl,responseData->responseBuffer,responseData->contentSize);
+    if(!responseData->responseBuffer) {
+      imageEventEmitter->onError();
```

Comment:
As this Piece of code is repeated in Three places on different scenario . For ONLoad - twice/OnError/. Suggest to have a common API  

      imageEventEmitter->onLoad();
      imageEventEmitter->onLoadEnd();
      hasToTriggerEvent_ = false;